xapian-core/weight/tfidfweight.cc

```diff
@@ -70,26 +74,48 @@ TfIdfWeight::name() const
 string
 TfIdfWeight::serialise() const
 {
-    return normalizations;
+    if (param_s > 0 || param_delta > 0) {
+	string result = serialise_double(param_s);
+	result += serialise_double(param_delta);
+	return result;
+    } else {
+	return normalizations;
+    }
 }
 
 TfIdfWeight *
-TfIdfWeight::unserialise(const string & s) const
+TfIdfWeight::unserialise(const string & str) const
 {
-    if (s.length() != 3)
-	throw Xapian::SerialisationError("Extra data in TfIdfWeight::unserialise()");
-    return new TfIdfWeight(s);
+    if (param_s > 0 || param_delta > 0) {
```

Comment:
>  A serialised version needs to be able to be serialised without knowing how it was built.
Oh, now I understand. It wouldn't be obvious for someone who doesn't know how serialisation was done to use how to unseriunserialise. Sorry, it looked so obvious to me. Do you any ideas about a good way of doing this? I can only think of writing a new method and use it to update parameter values at this point.